internal/subscriber/retry/metrics.go

```diff
@@ -0,0 +1,21 @@
+package retry
+
+import (
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	env                                string
+	subscriberTotalMessagesPushedToDLQ *prometheus.GaugeVec
+)
+
+func init() {
+	env = os.Getenv("APP_ENV")
```

Comment:
We have to refactor this part. Fetching Env from the underlying machine is an expensive call and we repeat this over every metric component across all packages. We should look at passing this down from the driver instead.

Reply:
Probably a refactor we should address in a separate task.